sharkfuser/include/fusilli/graph.h

```diff
@@ -213,6 +284,78 @@ Graph::convFProp(const std::shared_ptr<TensorAttr> &x,
   return y;
 }
 
+inline ErrorOr<Graph::CachedAssets>
+Graph::generateCompiledArtifacts(Backend backend,
+                                 const std::string &generatedAsm, bool remove) {
+  FUSILLI_LOG_LABEL_ENDL("INFO: Generating compiled artifacts");
```

Comment:
In a real scenario, we might have `g.generateCompiledArtifacts` run multiple times - I don't see us check in `Graph::generateCompiledArtifacts` if assets already exist and return them directly if so? If keeping them separate is preferred, perhaps create a wrapper method called `readOrGenerateCompiledArtifacts` that first tries to read and if not calls the generate method. We also need to test sequential invocations to ensure first time it misses and second time hits.

Reply:
Design-wise, my intent in this PR was for the two new methods to be low-level utilities that `g.execute` will someday call call. 

I'm picturing `g.execute` would go something like something like:

```c++
std::string asmText = TEST_FUSILLI_TRY(emitAsm());

runtime_module_type rmt;  // I made this up for the exmaple
if (hasCompiled) {
  rmt.set_vmfb(TEST_FUSILLI_TRY(readCompiledArtifacts(asmText).output.read()));
} else {
  rmt.set_vmfb(TEST_FUSILLI_TRY(generateCompiledArtifacts(asmText).output.read()));
  hasCompiled = true;
}
```

Personally, I like that the “read” and “generate” paths are separate, I think it might help us avoid any always-regenerate bugs.

Reply:
Sounds great!